### PR TITLE
HeapSort in Python

### LINKED ISSUE #648: Heap Sort
## 🚀 Feature

**Heap Sort Implementation**
-Heap sort is a comparison based sorting technique based on Binary Heap data structure. It is similar to selection sort where we first find the maximum element and place the maximum element at the end. We repeat the same process for remaining element.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

## Motivation

Heap sort is one of the basic sorting algorithm.  Its time complexity is O(nlogn).

## Pitch

Heap sort algorithm has limited uses because Quicksort and Mergesort are better in practice. Nevertheless, the Heap data structure itself is enormously used. Heap sort can be used in problems like where we need to sort almost sorted array and so on.

## Languages Supported:
- C [@rgrupesh]
- CPP [@bagladivyang03]
- Python [@avinashkranjan]
- Java [@ritvij14 ]
- JavaScript [@nagasuruthika]
- C# 
- GO [@GajjarMiten ]

### PR DESCRIPTION
**Pull request by avinashkranjan**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Heap Sort Implementation
-Heapsort is a comparison based sorting technique based on a Binary Heap data structure. It is similar to selection sort where we first find the maximum element and place the maximum element at the end. We repeat the same process for the remaining element.

- Python

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

### CODE REVIEW

File: Python/sort/Heap_sort.py [hunk 1]
```diff
@@ -1,19 +1,66 @@
...
+    # for each sorted heap, swap the root and the last number
+    for index in range(length - 1, 0, -1):
+        arr[index], arr[0] = arr[0], arr[index]
+        # then call heapify again with the new array
+        heapify(arr, index, 0)
+
+
+# Taking Elements to be Sorted
+data = [int(input())
+        for i in range(int(input("Enter elements to be sorted : ")))]
```
Comment (SKAUL05) [inline Python/sort/Heap_sort.py]: index in place of i

**Review by SKAUL05 — CHANGES_REQUESTED** [review]

File: Python/sort/Heap_sort.py (hunk 1 above)
Comment (avinashkranjan) [reply Python/sort/Heap_sort.py]: Done

**Review by atarax665 — APPROVED** [review]
LGTM :100:

**Review by iamrajiv — APPROVED** [review]

**Comment by avinashkranjan** [comment]
@iamrajiv  .. Take a Look here too..

**Review by iamrajiv — APPROVED** [review]

**Review by SKAUL05 — APPROVED** [review]

**Comment by iamrajiv** [comment]
@all-contributors please add @avinashkranjan for code

**Pull request merged**